src/hotspot/share/compiler/compileBroker.cpp

```diff
@@ -1023,6 +1023,20 @@ void CompileBroker::init_compiler_threads() {
 
 void CompileBroker::possibly_add_compiler_threads(JavaThread* THREAD) {
 
+  int old_c2_count = 0, new_c2_count = 0, old_c1_count = 0, new_c1_count = 0;
+  const int c2_tasks_per_thread = 2, c1_tasks_per_thread = 4;
+
+  // Do a quick check first without taking the lock. The later ones are more expensive.
+  if (_c2_compile_queue != nullptr) {
+    old_c2_count = _compilers[1]->num_compiler_threads();
```

Comment:
Can you use accessors here `get_c1_thread_count()` and `get_c2_thread_count()` ?